ewatercycle/models/pcrglobwb.py

```diff
@@ -11,112 +11,195 @@
+
+        version: pick a version from :py:attr:`~available_versions`
+        parameter_set: instance of :py:class:`~PCRGlobWBParameterSet`.
+        forcing: ewatercycle forcing container;
+            see :py:mod:`ewatercycle.forcing`.
+
+    Attributes:
+
         bmi (Bmi): GRPC4BMI Basic Modeling Interface object
     """
+    available_versions = ('setters', )
+
+    def __init__(self, version: str, parameter_set: PCRGlobWBParameterSet,
+                 forcing: PCRGlobWBForcing):
+        super().__init__()
+
+        self.version = version
+        self.parameter_set = parameter_set
+        self.forcing = forcing
+
+        self._set_docker_image()
+
+        self._setup_work_dir()
+        self._setup_default_config()
+
+    def _set_docker_image(self):
+        images = {
+            "setters": "ewatercycle/pcrg-grpc4bmi:setters",
+        }
+        self.docker_image = images[self.version]
+
+    def _setup_work_dir(self):
+        # Must exist before setting up default config
+        timestamp = time.strftime("%Y%m%d_%H%M%S")
+        work_dir = Path(CFG["output_dir"]) / f'pcrglobwb_{timestamp}'
+        work_dir.mkdir()
+        self.work_dir = work_dir.expanduser().resolve()
+
+    def _setup_default_config(self):
+        config_file = self.parameter_set.default_config
+        input_dir = self.parameter_set.input_dir
+
+        cfg = CaseConfigParser()
+        cfg.read(config_file)
+        cfg.set('globalOptions', 'inputDir', str(input_dir))
+        cfg.set('globalOptions', 'outputDir', str(self.work_dir))
+        cfg.set('globalOptions', 'startTime',
+                get_time(self.forcing.start_time).strftime("%Y-%m-%d"))
+        cfg.set('globalOptions', 'endTime',
+                get_time(self.forcing.start_time).strftime("%Y-%m-%d"))
+        cfg.set(
+            'meteoOptions', 'temperatureNC',
+            str((Path(self.forcing.directory) /
+                 self.forcing.temperatureNC).expanduser().resolve()))
+        cfg.set(
+            'meteoOptions', 'precipitationNC',
+            str((Path(self.forcing.directory) /
+                 self.forcing.precipitationNC).expanduser().resolve()))
+
+        self.config = cfg
+
     def setup(  # type: ignore
             self,
-            input_dir: PathLike,
-            cfg_file: PathLike,
-            additional_input_dirs: Iterable[PathLike] = [],
+            additional_input_dirs: Optional[Iterable[str]] = None,
             **kwargs) -> Tuple[PathLike, PathLike]:
         """Start model inside container and return config file and work dir.
 
         Args:
+            additional_input_dirs: By default, the model only has access to
+                its working directory, the parameter sets, and the forcing
+                directory. This option makes it possible to add other external
+                inputs, that can then be configured in the config file.
+            **kwargs: Use :py:meth:parameters to see all configurable options for this model,
```

Comment:
Instead of pointing to parameters it would be nice to spell out the available arguments.
So have `start_time, end_time, routing_method, dynamic_flood_plain, max_spinups_in_years` as setup arguments with docstrings. Then it will be possible for example to document which routing methods are possible (accuTravelTime|kinematicWave) .
We can also do this later.

Fix link in Sphinx
```suggestion
            **kwargs: Use :py:meth:`parameters` to see all configurable options for this model,
```

Reply:
Let's do that later, since I'm not sure what to document for these options, and this is probably not the definitive set anyway.

Reply:
Later at https://dev.azure.com/nlesc-team-beta/Team%20Beta/_boards/board/t/Team%20Beta/Backlog%20items/?workitem=1625